packages/plugins/i18n/admin/src/components/CMEditViewLocalePicker.tsx

```diff
@@ -0,0 +1,210 @@
+import { SingleSelect, SingleSelectOption, SingleSelectProps } from '@strapi/design-system';
+import {
+  Permission,
+  pxToRem,
+  useCMEditViewDataManager,
+  useQueryParams,
+} from '@strapi/helper-plugin';
+import { stringify } from 'qs';
+import { useIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Localization } from '../utils/data';
+import { getTranslation } from '../utils/getTranslation';
+
+import type { Locale } from '../store/reducers';
+import type { Entity } from '@strapi/types';
+
+export interface CMEditViewLocalePickerProps {
+  appLocales: Locale[];
+  createPermissions: Permission[];
+  currentEntityId: Entity.ID;
+  currentLocale: string;
+  localizations: Localization[];
+  readPermissions: Permission[];
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * CMEditViewLocalePicker
+ * -----------------------------------------------------------------------------------------------*/
+
+const CMEditViewLocalePicker = ({
+  appLocales = [],
+  createPermissions = [],
+  currentEntityId,
+  currentLocale,
+  localizations = [],
+  readPermissions = [],
+}: CMEditViewLocalePickerProps) => {
+  const { formatMessage } = useIntl();
+  const { hasDraftAndPublish, isSingleType, slug } = useCMEditViewDataManager();
+
+  interface Query {
+    plugins: {
+      i18n: {
+        locale: string;
+        relatedEntityId?: Entity.ID;
```

Comment:
Since this comes from the URL it would always be a string I suppose?

Reply:
You'd think that, the issue is that we can't assign IDs to it because they're either number or string 🤪